src/components/Picker/types.ts

```diff
@@ -0,0 +1,94 @@
+import {ChangeEvent, Component, ReactElement} from 'react';
+import {MeasureLayoutOnSuccessCallback, NativeMethods, StyleProp, ViewStyle} from 'react-native';
+
+type MeasureLayoutOnFailCallback = () => void;
+
+type RelativeToNativeComponentRef = (Component<unknown> & Readonly<NativeMethods>) | number;
+
+type BasePickerHandle = {
+    focus: () => void;
+    measureLayout: (relativeToNativeComponentRef: RelativeToNativeComponentRef, onSuccess: MeasureLayoutOnSuccessCallback, onFail: MeasureLayoutOnFailCallback) => void;
+};
+
+type OnMouseDown = () => void;
+
+type OnChange<TPickerValue> = (value: TPickerValue, index: number) => void;
+
+type AdditionalPickerEvents = {
+    onMouseDown?: OnMouseDown;
+    onChange?: (event: ChangeEvent<HTMLSelectElement>) => void;
+};
+
+type AdditionalPickerEventsCallback<TPickerValue> = (onMouseDown: OnMouseDown, onChange: OnChange<TPickerValue>) => AdditionalPickerEvents;
+
+type DefaultPickerEventsCallback = () => void;
+
+type PickerSize = 'normal' | 'small';
+
+type PickerItem<TPickerValue> = {
+    /** The value of the item that is being selected */
+    value: TPickerValue;
+
+    /** The text to display for the item */
+    label: string;
+};
+
+type PickerPlaceholder = {
+    /** The value of the placeholder item, usually an empty string */
+    value?: string;
+
+    /** The text to be displayed as the placeholder */
+    label?: string;
+};
+
+type BasePickerProps<TPickerValue> = {
+    /** BasePicker label */
+    label?: string | null;
+
+    /** Should the picker appear disabled? */
+    isDisabled?: boolean;
+
+    /** Input value */
+    value?: TPickerValue | null;
+
+    /** The items to display in the list of selections */
+    items: Array<PickerItem<TPickerValue>>;
+
+    /** Something to show as the placeholder before something is selected */
+    placeholder?: PickerPlaceholder;
+
+    /** Error text to display */
+    errorText?: string;
+
+    /** Customize the BasePicker container */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Customize the BasePicker background color */
+    backgroundColor?: string;
+
+    /** The ID used to uniquely identify the input in a Form */
+    inputID?: string;
```

Comment:
```suggestion
    inputID?: string;

    /** Saves a draft of the input value when used in a form */
    shouldSaveDraft?: boolean,
```

Although is not being directly used, `shouldSaveDraft` is used dynamically by `Form` component when dealing with its children, so we must keep it.